privaterelay/context_processors.py

```diff
@@ -77,3 +50,44 @@ def _get_fxa(request):
         return fxa
     except AttributeError:
         return None
+
+
+@lru_cache(maxsize=None)
+def _get_csat_cookie_and_reason(request):
+    if request.user.is_anonymous:
+        return '', None
+
+    day_periods = [7, 30, 90]
+    profile = request.user.profile_set.first()
+    first_visit = request.COOKIES.get("first_visit")
+    reasons = ['free', 'premium']
+    days_since_args = {
+        'free': first_visit,
+        'premium': profile,
+    }
+
+    for reason in reasons:
+        days_since = _get_days_since(days_since_args[reason])
+        if reason == 'premium' and days_since == None:
+            return '', None
+        for num_days in day_periods:
+            if (days_since >= num_days):
+                csat_dismissal_cookie = (
+                    f'csat-survey-{reason}-{num_days}days_dismissed'
+                )
+                if (request.COOKIES.get(csat_dismissal_cookie)):
+                    return csat_dismissal_cookie, None
```

Comment:
Dang, this `return` will always execute as soon as someone dismisses the `free7days` or the `premium7days` prompt. We'll never go into the next iteration over the `day_periods` loop. Needs some re-thinking again.